File: rock_paper_scissors.py
import sys, os
from random import randint

while True:
    OPTIONS = {1:"rock", 2:"paper", 3:"scissors"}
    score_player = 0
    score_computer = 0

    def did_player_win(player_move, computer_move):

        if OPTIONS[player_move] == "rock" and computer_move != "paper":
            return True
        else:
            return False
            
        if OPTIONS[player_move] == "paper" and computer_move != "rock":
            return True
        else:
            return False
            
        if OPTIONS[player_move] == "scissors" and computer_move != "rock":
            return True
        else:
            return False
    
    def menu():
        os.system('cls')
        print("Welcome to ROCK_PAPER_SCISSORS!\n")
        
        if score_player > 0 or score_computer > 0:
            print("\tCurrent Scores")
            print("\t----------------")
            print("\tPlayer  : {}".format(score_player))
            print("\tComputer: {}\n".format(score_computer))
        
        for el in OPTIONS:
            print(str(el)+'.', OPTIONS[el])
            
        the_game()

    def the_game():
        global score_computer, score_player
        while True:
            try:
                player_move = int(input("\nMake your move! >> "))
                if not player_move > len(OPTIONS) and not player_move < 1:
                    break
                else:
                    raise ValueError
            except ValueError:
                print('\n\n!ERROR : Please select a number from the menu.')
                continue   
        
        computer_move = OPTIONS[randint(1,3)]

        print("\nYou chose {}".format(OPTIONS[player_move]))
        print("The computer chose {}".format(computer_move))
        
        if OPTIONS[player_move] == computer_move:
            print("\nDRAW!")
        else:
            if did_player_win(player_move, computer_move):
                score_player += 1
                print("\nYou WIN!")
            else:
                score_computer += 1
                print("\nYou LOSE!")
        
        play_again = input("\nWould you like to play again? >> ")
        if play_again.lower() == 'y' or play_again.lower() == "yes":
            menu()
        else:
            raise SystemExit()
    
    menu()

File: java/rockPaperScissors.java
import java.util.Scanner;
import java.lang.Math;
import java.util.HashMap;
import java.util.Random;

class Main {
  
  private int pScore = 0;
  private int cScore = 0;
  private boolean gameRunning = true;
  
  private void showScore() {
    System.out.println("\tCurrent Scores");
    System.out.println("\t----------------");
    System.out.println("\tPlayer  : " + pScore);
    System.out.println("\tComputer: "+ cScore + "\n");
  }

  private void showMenu() {
    System.out.println("WELCOME TO rockPaperScissors!");
    if (pScore > 0 || cScore > 0) {
      showScore();
    }
    System.out.println("1. rock");
    System.out.println("2. paper");
    System.out.println("3. scissors");          
  }

  private String getMove(int move) {
    HashMap<Integer, String> OPTIONS = new HashMap<>();
    OPTIONS.put(1, "rock");
    OPTIONS.put(2, "paper");
    OPTIONS.put(3, "scissors");
    return OPTIONS.get(move);
  }

  private boolean win(String playerMove, String compMove) {
    if (playerMove == "rock" && compMove != "paper") {
      return true;
    } else if (playerMove == "paper" && compMove != "rock") {
      return true;
    } else if (playerMove == "scissors" && compMove != "rock") {
      return true;
    } else {
      return false;
    }
  }

  private void round() {
    Scanner sc = new Scanner(System.in);
    System.out.println("\nMake your move! >> ");
    String playerMove = getMove(sc.nextInt());
    Random rand = new Random();
    String computerMove = getMove(rand.nextInt(3) + 1);
    
    System.out.println("\nYou chose: " + playerMove);
    System.out.println("Computer chose: " + computerMove);

    if (win(playerMove, computerMove)) {
      System.out.println("\nYou win!");
      pScore++;
    } else { 
      System.out.println("\nComputer wins!");
      cScore++;
    }
  }

  private void playAgain() {
    Scanner sc = new Scanner(System.in);
    System.out.println("Continue? [y/n] >> ");
    char response = sc.nextLine().toLowerCase().charAt(0);
    if (response == 'n') {
      showScore();
      gameRunning = false;
    }
  }

  public void run() {
    while (gameRunning == true) {
      showMenu();
      round();
      playAgain();
    }
  }

  public static void main(String[] args) { 
    Main game = new Main();
    game.run();
  }
  
}
